test/e2e/framework/framework.go

```diff
@@ -69,7 +76,17 @@ type Framework struct {
 
 // NewDefaultFramework constructs a new e2e test Framework with default options.
 func NewDefaultFramework(baseName string) *Framework {
-	return NewFramework(baseName, nil)
+	f := NewFramework(baseName, nil)
+	f.SkipInitCloudProvider = true
+	return f
+}
+
+// NewFrameworkWithCloudProvider constructs a new e2e test Framework for testing
+// cloudprovider.Interface directly.
+func NewFrameworkWithCloudProvider(baseName string) *Framework {
+	f := NewFramework(baseName, nil)
+	f.SkipNamespaceCreation = true
```

Comment:
why is this and skip init true?

Reply:
This is where the two differing test approaches mashed into one test framework breaks down a little. For Zones and Instances I'm testing via the cloudprovider.Interface and for LoadBalancer I ported over the framework code for creating the namespace. Any ideas for improvement?